clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java

```diff
@@ -198,6 +196,73 @@ public void testExists_NotExists() throws ApiException, IOException {
 
         Assert.assertFalse(fs.exists(p));
     }
+    
+    @Test
+    public void testDelete_FileExists() throws ApiException, IOException {
+        when(objectsApi.statObject(eq("repo"), eq("main"), eq("delete/sample/file.txt")))
+                .thenReturn(new ObjectStats().
+                        path("lakefs://repo/main/delete/sample/file.txt").
+                        pathType(PathTypeEnum.OBJECT).
+                        physicalAddress(s3Url("delete/sample/file.txt")).
+                        checksum("unused").
+                        mtime(0L).
+                        sizeBytes(42L));
+        // return true because file found
+        Assert.assertTrue(
+                fs.delete(new Path("lakefs://repo/main/no/place/file.txt"), false));
+    }
+
+    @Test
+    public void testDelete_FileNotExists() throws ApiException, IOException {
+        doThrow(new ApiException(HttpStatus.SC_NOT_FOUND, "not found"))
+                .when(objectsApi).deleteObject(eq("repo"), eq("main"), eq("no/place/file.txt"));
```

Comment:
We use the opposite style here, mostly because that's what the Mockito tutorial shows.  I don't mind either order, but let's pick one set of verbs and use it.
```suggestion
        when(objectsApi).deleteObject("repo", "main", "no/place/file.txt")
            .thenThrow(new ApiException(HttpStatus.SC_NOT_FOUND, "not found"));
```

Reply:
didn't work for me - had no choice on how to handle this case: mock a method that returns void.

Reply:
Right... I forgot that you have to do this because Java.